libs/SalesforceSDKCommon/SalesforceSDKCommon/Classes/Keychain/KeychainManager.swift

```diff
@@ -465,56 +465,76 @@ public class KeychainHelper: NSObject {
     }
     
     internal class CachedWrapper {
-        
-        static var cache: SafeMutableDictionary<NSString, KeychainResult> = SafeMutableDictionary<NSString, KeychainResult>()
-        
+        static private(set) var cache = Dictionary<NSString, KeychainResult>()
+        static private var queue = DispatchQueue(label: "com.salesforce.mobilesdk.readWriteQueue\(arc4random_uniform(32))", qos: .unspecified, attributes: [.concurrent], autoreleaseFrequency: .inherit, target: nil)
+
         class func key(service: String, account: String?) -> NSString {
             guard let acc = account else {
                 return NSString(string: "\(service)")
             }
             return NSString(string: "\(service)_\(acc)")
         }
         
-        class func wrap(_ service: String, _ account: String?, readFunc: (String, String?) -> KeychainResult ) -> KeychainResult {
-            guard let result = cache[key(service: service, account: account)]  else{
-                let keychainResult =  readFunc(service, account)
-                if keychainResult.success {
-                    cache[key(service: service, account: account)]  = keychainResult
+        class func wrap(_ service: String, _ account: String?, readFunc: (String, String?) -> KeychainResult) -> KeychainResult {
```

Comment:
For now I made a similar change as #3443 but I’d like to make it more generic, unify them, and add the rest of the operations so it could be the swift rewrite of SFSDKSafeMutableDictionary (maybe for 10.1?)

Reply:
Sounds good.